src/io/flutter/run/test/FlutterTestRunner.java

```diff
@@ -62,9 +71,64 @@ public boolean canRun(@NotNull String executorId, @NotNull RunProfile profile) {
   @Override
   protected RunContentDescriptor doExecute(@NotNull RunProfileState state, @NotNull ExecutionEnvironment env)
     throws ExecutionException {
-    return runInDebugger((TestLaunchState)state, env);
+    if (env.getExecutor().getId().equals(ToolWindowId.RUN)) {
+      return run((TestLaunchState)state, env);
+    } else {
+      return runInDebugger((TestLaunchState)state, env);
+    }
   }
 
+  protected RunContentDescriptor run(@NotNull TestLaunchState launcher, @NotNull ExecutionEnvironment env)
+    throws ExecutionException {
+    final ExecutionResult executionResult = launcher.execute(env.getExecutor(), this);
+    final ObservatoryConnector connector = new Connector(executionResult.getProcessHandler());
+
+    ApplicationManager.getApplication().executeOnPooledThread(() -> {
+      // Poll, waiting for "flutter run" to give us a websocket.
+      // This is adapted from DartVmServiceDebugProcess::scheduleConnect
+      String url = connector.getWebSocketUrl();
+
+      while (url == null) {
+        TimeoutUtil.sleep(100);
+        url = connector.getWebSocketUrl();
+      }
+
+      final VmService vmService;
+      try {
+        vmService = VmService.connect(url);
```

Comment:
Perhaps add a comment indicating that we're looking to resume any paused-at-start isolates here.